plugin/plugin.go

```diff
@@ -0,0 +1,362 @@
+	CreateSandbox           bool   `envconfig:"PLUGIN_CREATE_SANDBOX"`
+	Debug                   bool   `envconfig:"PLUGIN_DEBUG"`
+	UploadIncludesPattern   string `envconfig:"PLUGIN_UPLOAD_INCLUDES_PATTERN"`
+	UploadExcludesPattern   string `envconfig:"PLUGIN_UPLOAD_EXCLUDES_PATTERN"`
+	ScanIncludesPattern     string `envconfig:"PLUGIN_SCAN_INCLUDES_PATTERN"`
+	ScanExcludesPattern     string `envconfig:"PLUGIN_SCAN_EXCLUDES_PATTERN"`
+	FileNamePattern         string `envconfig:"PLUGIN_FILE_NAME_PATTERN"`
+	ReplacementPattern      string `envconfig:"PLUGIN_REPLACEMENT_PATTERN"`
+	ScanAllNonFatalTopLevel bool   `envconfig:"PLUGIN_SCAN_ALL_NON_FATAL_TOP_LEVEL_MODULES"`
+	IncludeNewModules       bool   `envconfig:"PLUGIN_INCLUDE_NEW_MODULES"`
+	PHost                   string `envconfig:"PLUGIN_P_HOST"`
+	PPort                   string `envconfig:"PLUGIN_P_PORT"`
+	PUser                   string `envconfig:"PLUGIN_P_USER"`
+	PPassword               string `envconfig:"PLUGIN_P_PASSWORD"`
+	VID                     string `envconfig:"PLUGIN_VID"`
+	VKey                    string `envconfig:"PLUGIN_VKEY"`
+	DeleteIncompleteScan    bool   `envconfig:"PLUGIN_DELETE_INCOMPLETE_SCAN"`
+	WaitForScan             bool   `envconfig:"PLUGIN_WAIT_FOR_SCAN"`
+	TimeoutFailsJob         bool   `envconfig:"PLUGIN_TIMEOUT_FAILS_JOB"`
+	CanFailJob              bool   `envconfig:"PLUGIN_CAN_FAIL_JOB"`
+	UseProxy                bool   `envconfig:"PLUGIN_USE_PROXY"`
+	Version                 string `envconfig:"PLUGIN_VERSION"`
+	Level                   string `envconfig:"PLUGIN_LEVEL"`
+}
+
+func ValidateInputs(args Args) error {
+	if args.AppName == "" {
+		return fmt.Errorf("PLUGIN_APPLICATION_NAME is required")
+	}
+	if args.VID == "" || args.VKey == "" {
+		return fmt.Errorf("PLUGIN_VID and PLUGIN_VKEY must be provided")
+	}
+	if args.UploadIncludesPattern == "" && args.UploadExcludesPattern == "" {
+		args.UploadIncludesPattern = "**/*"
+	}
+	return nil
+}
+
+func Exec(ctx context.Context, args Args) error {
+	logrus.Infof("🟢 Starting Veracode UploadAndScan")
+
+	finalFileList, err := resolveUploadFileList(args.UploadIncludesPattern, args.UploadExcludesPattern)
+	if err != nil {
+		return fmt.Errorf("failed to resolve upload file list: %w", err)
+	}
+
+	cmdArgs := buildVeracodeCommandArgs(args, finalFileList)
+	logrus.Infof("➡️  Executing: java %s", strings.Join(cmdArgs, " "))
+
+	cmd := exec.CommandContext(ctx, "java", cmdArgs...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+
+	err = runJavaCommandWithTimeout(cmd, args)
+	if err != nil {
+		return err
+	}
+
+	if args.WaitForScan {
+		return waitForScanCompletion(ctx, args)
+	}
+
+	return nil
+}
+
+func buildVeracodeCommandArgs(args Args, fileList string) []string {
+	cmdArgs := []string{
+		"-jar", "/opt/veracode/api-wrapper.jar",
+		"-action", "UploadAndScan",
+		"-appname", args.AppName,
+		"-filepath", fileList,
+		"-version", args.Version,
+		"-vid", args.VID,
+		"-vkey", args.VKey,
```

Comment:
```suggestion
		"-vkey", args.VKey,
```
this is printed as plain text in the logs, either use env variables or mask this input. Follow the same in  isScanPublished and handlePolicyEvaluation accordingly.